### PR TITLE
789 Display post cert milsetones and conditionally display milestone dates

### PR DESCRIPTION
**Pull request by godfreyyeung**
1. Two milestones (Pre and Post Hearing) weren't rendered, so we add them to the list of milestones to retrieve from CRM.
2. We include these two milestones and also the CPC Review milestone as those which should render the "review meeting date" instead of actual start/end date.
3. We hide dates under milestones until milestone has statuscode of "In Progress" or "Completed"

Fixes [AB#13608](https://dcp-paperless.visualstudio.com/d36fd830-9029-4b77-b0c4-b0df2012eb98/_workitems/edit/13608)

### CODE REVIEW

File: client/app/templates/components/project-milestone.hbs [hunk 1]
```diff
@@ -40,19 +42,21 @@
...
-      {{#if milestone.displayDate2}}
-        &ndash;
-        <DateDisplay
-          @date={{milestone.displayDate2}}
-          @outputFormat="MMMM D, YYYY"
-        />
-      {{/if}}
-      {{!-- {{#if timeRelativeToNow}}
-        <div class="milestone-offset">{{timeRelativeToNow}}</div>
-      {{/if}} --}}
```
Comment (trbmcginnis) [inline client/app/templates/components/project-milestone.hbs:53]: Thanks for cleaning this up!

File: client/app/templates/components/project-milestone.hbs [hunk 2]
```diff
@@ -26,9 +26,11 @@
       </sup>
     {{/if}}
   </h4>
+
   {{#if milestone.displayDate}}
```
Comment (trbmcginnis) [inline client/app/templates/components/project-milestone.hbs:30]: It looks like `{{#if milestone.displayDate2}}` is within the `{{#if (in-array (array "In Progress" "Completed") milestone.statuscode)}}`

Should `{{#if milestone.displayDate}}` also be within that if?

File: client/tests/acceptance/hearings-list-for-milestones-list-shows-up-correctly-test.js [hunk 3]
```diff
@@ -493,8 +493,8 @@ module('Acceptance | hearings list for milestones list shows up correctly', func
     assert.ok(this.element.querySelector('[data-test-milestone-name="38"]').textContent.includes('Borough Board Review'), 'BB milestone');
 
     assert.ok(this.element.querySelector('[data-test-milestone-dates="1"]').textContent.includes, 'July 13, 2019');
-    assert.ok(this.element.querySelector('[data-test-milestone-dates="11"]').textContent.includes, 'October 18, 2019');
-    assert.ok(this.element.querySelector('[data-test-milestone-dates="12"]').textContent.includes, 'October 5 - October 29, 2019');
+    assert.notOk(this.element.querySelector('[data-test-milestone-dates="11"]'));
+    assert.notOk(this.element.querySelector('[data-test-milestone-dates="12"]'));
```
Comment (trbmcginnis) [inline client/tests/acceptance/hearings-list-for-milestones-list-shows-up-correctly-test.js:497]: Cool, nice transition from the older tests :)

**Review by trbmcginnis — APPROVED** [review]
Looks great, thanks!

File: client/app/templates/components/project-milestone.hbs (hunk 2 above)
Comment (godfreyyeung) [reply client/app/templates/components/project-milestone.hbs:30]: I don't think so. It looks like the outer  `{{#if milestone.displayDate}}` is used to conditionally display milestone links and outcomes. The new requirement to add `{{#if (in-array (array "In Progress" "Completed") milestone.statuscode)}}` only mentioned it is for conditionally displaying/hiding the dates, not the milestone links and outcomes

**Pull request merged**